core/pkg/ingress/types.go

```diff
@@ -40,6 +41,10 @@ var (
 // Controller holds the methods to handle an Ingress backend
 // TODO (#18): Make sure this is sufficiently supportive of other backends.
 type Controller interface {
+	// HealthzChecker returns is a named healthz check that returns the ingress
+	// controller status
+	healthz.HealthzChecker
```

Comment:
actually i prefer giving this interface a plain Check method, the name "healthz" implies a http endpoint and the Check method call may do anything else to validate health